### PR TITLE
✨ forwardRef wrapped in a memo now works with type inference

### LINKED ISSUE #162: Feat: Add support for forwardRef used within memo using TS type inference
❌ Bad, type inference for `ref` is lost because memo overrides it
```jsx
const MyComponent = memo<MyComponentProps>(forwardRef<HTMLElement, MyComponentProps>((props, ref) => {}));
```
🔥 
```jsx
<MyComponent ref={foo} />
```

✅  Good, `ref` is inferred 
```jsx
const MyComponent = memo(forwardRef<HTMLElement, MyComponentProps>((props, ref) => {}));
```
😄  
```jsx
<MyComponent ref={foo} />
```

### PR DESCRIPTION
**Pull request by danieldelcore**
Closes #162

### CODE REVIEW

File: packages/extract-react-types/converters-typescript.test.js [hunk 1]
```diff
@@ -601,6 +601,23 @@ const TESTS = [
...
+    name: 'memo wrapping forwardRef, with inferred types',
+    typeSystem: 'typescript',
+    code: `
+    import React, { forwardRef, memo } from 'react';
+
+    type MyComponentProps = {
+      foo: string,
+    }
+
+    const MyComponent = memo(forwardRef<HTMLElement, MyComponentProps>((props, ref) => {
```
Comment (pgmanutd) [inline packages/extract-react-types/converters-typescript.test.js:615]: Will `forwardRef` also works without generics? Like this:

```ts
const MyComponent = memo(forwardRef((props: MyComponentProps, ref: React.Ref<HTMLElement>) => {
```

Comment (danieldelcore) [reply packages/extract-react-types/converters-typescript.test.js:615]: Hey! Yeah that should work, there's a test case for this above I'm pretty sure 😀

Comment (danieldelcore) [reply packages/extract-react-types/converters-typescript.test.js:615]: Might be in the flow tests actually

Comment (pgmanutd) [reply packages/extract-react-types/converters-typescript.test.js:615]: Nice. @madou: You have any comments? Looks good to me overall.

**Review by madou — APPROVED** [review]
lgtm

**Pull request merged**